packages/roosterjs-editor-api/lib/format/setAlignment.ts

```diff
@@ -1,36 +1,146 @@
 import execCommand from '../utils/execCommand';
+import { VTable } from 'roosterjs-editor-dom';
 import {
     Alignment,
     ChangeSource,
     DocumentCommand,
+    ExperimentalFeatures,
     IEditor,
     QueryScope,
+    SelectionRangeTypes,
+    TableSelectionRange,
 } from 'roosterjs-editor-types';
 
+const TABLE = 'TABLE';
+const TEXT = 'TEXT';
+
 /**
  * Set content alignment
  * @param editor The editor instance
  * @param alignment The alignment option:
  * Alignment.Center, Alignment.Left, Alignment.Right
  */
 export default function setAlignment(editor: IEditor, alignment: Alignment) {
-    let command = DocumentCommand.JustifyLeft;
-    let align = 'left';
+    const element = editor.getElementAtCursor();
+    const elementType = isATableOrText(element, editor);
+
+    editor.addUndoSnapshot(() => {
+        alignElement(editor, element, elementType, alignment);
+        editor.queryElements('[align]', QueryScope.OnSelection, node =>
+            alignElement(editor, node, elementType, alignment, true /** addUndoSnapshot */)
+        );
+    }, ChangeSource.Format);
+}
+
+/**
+ * Check if the element at the cursor is a table or text element
+ * @param element
+ * @returns
+ */
+function isATableOrText(element: HTMLElement, editor: IEditor) {
+    if (!element) {
+        return;
+    }
+    const selection = editor.getSelectionRangeEx();
+    const selectionType = selection.type;
+    if (selection && selectionType === SelectionRangeTypes.Normal) {
+        return TEXT;
+    } else if (
+        editor.isFeatureEnabled(ExperimentalFeatures.TableAlignment) &&
+        selection &&
+        selectionType === SelectionRangeTypes.TableSelection &&
+        isWholeTableSelected(selection)
+    ) {
+        return TABLE;
+    }

```

Comment:
else?

You can simplify your function to be:

return (type == table && isFeatureEnabled && isWholeTableSelected);

Making it returning a boolean should be easier to use